spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -267,4 +270,30 @@ public boolean aggregatePushDownEnabled() {
         .defaultValue(SparkSQLProperties.AGGREGATE_PUSH_DOWN_ENABLED_DEFAULT)
         .parse();
   }
+
+  public boolean distributedPlanningEnabled() {
+    return dataPlanningMode() != LOCAL || deletePlanningMode() != LOCAL;
+  }
+
+  public PlanningMode dataPlanningMode() {
+    String modeName =
+        confParser
+            .stringConf()
+            .sessionConf(SparkSQLProperties.PLANNING_MODE)
```

Comment:
Shall add an options parameter as well?

Reply:
Yeah, why not?

Reply:
Is there a use case for it, or is this speculative?

Reply:
Mostly speculative I guess if a query touches multiple tables but we want to enable that only in one?

Reply:
Yeah, that provides the ability for fine controls when there are multiple tables.